### PR TITLE
PRC4 Game Node API

### PR DESCRIPTION
**Pull request by acedward**
Game Node Dex API

    `GET /dex/`
    `GET /dex/{asset}`
    `GET /dex/{asset}/wallet/{wallet}`
    `GET /dex/{asset}/orders?seller=wallet&page=number&limit=number`
    `GET /dex/{asset}/historical_price?freq=string&start=number&end=number`

### CODE REVIEW

File: PRCS/prc-4.md [hunk 1]
```diff
@@ -150,6 +150,117 @@ It's not an AMM. In a typical AMM dex, there are liquidity pools and you trade a
...
+    }
+    ```
+
+
+3. Get ERC1155 tokens of `asset` for the specified `wallet` that have been minted or owned in a valid way. This is used by the DEX to get the list of valid assets that user is able to create sell orders for.
+
+    `GET /dex/{asset}/wallet/{wallet}`
+
+    * asset: valid name for specific game asset token.
+    * wallet_address: wallet to query for asset
```
Comment (matejos) [inline PRCS/prc-4.md]:
```suggestion
    * wallet: wallet to query for asset
```

File: PRCS/prc-4.md [hunk 2]
```diff
@@ -150,6 +150,117 @@ It's not an AMM. In a typical AMM dex, there are liquidity pools and you trade a
...
+            version?: string       // Optional Game Version
+        }
+    }
+    ```
+
+2. Get Asset information.
+
+    `GET /dex/{asset}`
+
+    * asset: valid name for specific game asset token.
```
Comment (matejos) [inline PRCS/prc-4.md:218]: I assume this corresponds with the `code` in `/dex/`->`assets`, correct?

File: PRCS/prc-4.md [hunk 3]
```diff
@@ -150,6 +150,117 @@ It's not an AMM. In a typical AMM dex, there are liquidity pools and you trade a
...
+    `GET /dex/`
+
+    RESPONSE 
+    ```js
+    {
+        assets: {
+            code: string;          // Asset Code
+            description: string;   // Asset Description
+            fromSym: string;       // Name of base Asset  
+            toSym: string;         // Name of unit to convert
```
Comment (matejos) [inline PRCS/prc-4.md]: Could we also get the contract address for the asset and for the asset DEX? (Note: each asset must have its own DEX).
And would be also good to have asset `name` (or is that what you mean by `description`?).
What I imagined:
```
{
code: 'tgold',
name: 'Tarochi Gold',
description: 'Main currency of the RPG game Tarochi',
fromSym: 'TGOLD',
toSym: 'ETH',
address: '0x...',
dexAddress: '0x...',
...
}
```

File: PRCS/prc-4.md (hunk 2 above)
Comment (acedward) [reply PRCS/prc-4.md:218]: this is to query specific info about assets, initially, it was for totalSupply only, but it's a good place to aggregate any info about the asset

```
GET /dex/tgold
Returns { totalSupply: 1234567890 }
```

Comment (acedward) [reply PRCS/prc-4.md:218]: the assets names can be obtained from `/dex` 
the only endpoint that doesn't require to specify an asset

Comment (acedward) [reply PRCS/prc-4.md:218]: Ahhh, I read this wrong.

Yes, it the `code` perhaps a better name as assetId and use `/dex/{assetId}` in the endpoints

**Review by matejos — APPROVED** [review]

**Pull request merged**